src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/MvcRazorPagesMvcCoreBuilderExtensions.cs

```diff
@@ -53,15 +53,25 @@ public static IMvcCoreBuilder AddRazorPages(this IMvcCoreBuilder builder)
         // Internal for testing.
         internal static void AddServices(IServiceCollection services)
         {
+            //
+            // Options
+            //
             services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<RazorPagesOptions>, RazorPagesOptionsSetup>());
 
+            //
+            // Action Invoker
+            //
             services.TryAddEnumerable(
                 ServiceDescriptor.Singleton<IActionDescriptorProvider, PageActionDescriptorProvider>());
+            services.TryAddSingleton<IActionDescriptorChangeProvider, PageActionDescriptorChangeProvider>();
 
             services.TryAddEnumerable(
                 ServiceDescriptor.Singleton<IActionInvokerProvider, PageActionInvokerProvider>());
 
+            //
```

Comment:
Does this commenting style match how we annotate things in other AddServices methods? If so we should make the comments below match, if not we should make these parts normal comments.